examples/simple_threshold_signature/threshold_signature.rs

```diff
@@ -0,0 +1,114 @@
+use crate::simple_threshold_signature::signature::Signature;
+use crate::Element;
+use alba::centralized_telescope::proof::Proof;
+use alba::centralized_telescope::Telescope;
+use blst::min_sig::{PublicKey, Signature as BlsSignature};
+use blst::BLST_ERROR;
+
+pub(crate) struct ThresholdSignature {
+    proof: Proof,
+}
+
+impl ThresholdSignature {
+    // Create the Alba proof. Convert signatures to bytes and use them as the prover set. Create Alba proof with the
+    // prover set. Collect signatures by converting proof elements to bls signatures. Find each signature's index and
+    // collect them in a list. Return alba proof and the indices.
+    pub(crate) fn aggregate(
+        signatures: &[Signature],
+        alba: &Telescope,
+        public_key_list: &[(usize, PublicKey)],
+    ) -> (Self, Vec<usize>) {
+        // Convert signatures to bytes and collect as the prover set.
+        let prover_set = signatures
+            .iter()
+            .map(|s| s.signature.to_bytes())
+            .collect::<Vec<Element>>();
+
+        println!("-- Creating alba proof. ");
+        // Create alba proof with the prover set
+        let proof = alba.prove(&prover_set).unwrap();
+        println!("-- Alba proof created: ");
+        println!(
+            " - Numbers of retries done to find the proof: {}",
+            proof.retry_counter
+        );
+        println!(
+            " - Index of the searched subtree to find the proof: {}",
+            proof.search_counter
+        );
+        println!(
+            " - Number of elements in the proof sequence: {}",
+            proof.element_sequence.len()
+        );
+
+        // Convert proof elements to signatures to obtain their indexes
+        let proof_signatures: Vec<BlsSignature> = proof
+            .element_sequence
+            .iter()
+            .filter_map(|element| BlsSignature::from_bytes(element).ok())
+            .collect();
+
+        // Collect the indices of the signatures that create alba proof.
+        let mut indices = Vec::with_capacity(proof_signatures.len());
+        for sig in &proof_signatures {
+            if let Some(signature_entry) = signatures.iter().find(|entry| entry.signature == *sig) {
+                if public_key_list
+                    .iter()
+                    .any(|entry| entry.0 == signature_entry.index)
```

Comment:
I think this logic wouldn't work if you pass signatures to this function that are not index 0, 1, ..., n_p-1. That's why in my document Aggregate gets triples (i, s_i, w_i). In particular, the index i of the signature is included.

Also, `public_key_list` seems redundant in this function.